### PR TITLE
TermvectorsResponse fix for optionals.

### PR DESCRIPTION
**Pull request by MikePieperSer**
### Description
This makes optional fields in TermvectorsResponse optional.

### Issues Resolved
Closes [#639 ]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: java-client/src/main/java/org/opensearch/client/opensearch/core/TermvectorsResponse.java [hunk 1]
```diff
@@ -157,8 +157,10 @@ protected void serializeInternal(JsonGenerator generator, JsonpMapper mapper) {
 		generator.writeKey("took");
 		generator.write(this.took);
 
-		generator.writeKey("_version");
-		generator.write(this.version);
+        if (null != this.version) {
```
Comment (dblock) [inline java-client/src/main/java/org/opensearch/client/opensearch/core/TermvectorsResponse.java:160]: Tabs vs. spaces looks suspicious here. Also surprised spotless didn't catch it, mind checking please?

Comment (dblock) [inline java-client/src/main/java/org/opensearch/client/opensearch/core/TermvectorsResponse.java:160]: Is this optional? Maybe add a test?

Comment (MikePieperSer) [reply java-client/src/main/java/org/opensearch/client/opensearch/core/TermvectorsResponse.java]: You have right. According to the server sources https://github.com/opensearch-project/OpenSearch/blob/7dc6683838951e1a211c6ff634d9eea93e2be3c2/server/src/main/java/org/opensearch/action/termvectors/TermVectorsResponse.java#L205 _version is still delivered.

**Comment by VachaShah** [comment]
@MikePieperSer Can you please fix the merge conflicts? Thank you!

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +30,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Fix version and build ([#254](https://github.com/opensearch-project/opensearch-java/pull/254))
 - Fix PutMappingRequest by removing unsupported fields ([#597](https://github.com/opensearch-project/opensearch-java/pull/597))
 - Fix parsing of GetFieldMappingResponse ([#641](https://github.com/opensearch-project/opensearch-java/pull/641))
+- [BUG] Fix TermvectorsResponse for optional fields. ([#642](https://github.com/opensearch-project/opensearch-java/pull/642))
```
Comment (dblock) [inline CHANGELOG.md:33]: Let's make this look like the other lines. Say `Fix ...`

**Review by dblock — CHANGES_REQUESTED** [review]
Super nit please.

File: CHANGELOG.md (hunk 2 above)
Comment (MikePieperSer) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md [hunk 3]
```diff
@@ -30,6 +30,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Fix version and build ([#254](https://github.com/opensearch-project/opensearch-java/pull/254))
 - Fix PutMappingRequest by removing unsupported fields ([#597](https://github.com/opensearch-project/opensearch-java/pull/597))
 - Fix parsing of GetFieldMappingResponse ([#641](https://github.com/opensearch-project/opensearch-java/pull/641))
+- Fix TermvectorsResponse for optional fields. ([#642](https://github.com/opensearch-project/opensearch-java/pull/642))
```
Comment (dblock) [inline CHANGELOG.md:33]: Remove the period please

**Review by dblock — APPROVED** [review]
You will hate me ;)

**Review by dblock — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Comment by VachaShah** [comment]
Not sure why the backport workflow did not run, I will look into it. Created a manual backport PR #658 for now.

**Pull request merged**